shared/src/main/scala/squants/Quantity.scala

```diff
@@ -306,5 +306,10 @@ abstract class Quantity[A <: Quantity[A]] extends Serializable with Ordered[A] {
    * @return
    */
   def map(f: Double ⇒ Double): A = unit(f(value))
+}
 
+abstract class StrictlyPositiveQuantity[A <: StrictlyPositiveQuantity[A]](value: Double) extends Quantity[A] {self: A ⇒
```

Comment:
This is interesting though I'd strongly prefer have a constructor returning an `Option[StrictlyPositiveQuantity[A]]` than throwing an exception here